pkg/frontend/validate.go

```diff
@@ -44,3 +44,20 @@ func (f *frontend) validateSubscriptionState(ctx context.Context, key string, al
 
 	return nil, api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidSubscriptionState, "", "Request is not allowed in subscription in state '%s'.", doc.Subscription.State)
 }
+
+// validateOpenShiftUniqueKey returns which unique key if causing a 412 error
+func (f *frontend) validateOpenShiftUniqueKey(ctx context.Context, doc *api.OpenShiftClusterDocument) error {
+	docs, err := f.db.OpenShiftClusters.QueryAll(ctx, doc.PartitionKey)
+	if err != nil {
+		return err
+	}
+	for _, existingDoc := range docs.OpenShiftClusterDocuments {
+		if existingDoc.ClusterResourceGroupIDKey == doc.ClusterResourceGroupIDKey {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidResourceGroup, "", "The provided resource group '%s' already contains a cluster.", doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID)
```

Comment:
Might be worth fixing `CloudErrorCodeInvalidResourceGroup` up to `CloudErrorCodeDuplicateResourceGroup` while you're in here?